app/routes/datasets/dataset.py

```diff
@@ -33,7 +33,7 @@ async def get_owner(
 
     dataset_row: ORMDataset = await datasets.get_dataset(dataset)
     owner: str = dataset_row.owner_id
-    if owner != user.id:
+    if owner != user.id and user.role != "ADMIN":
```

Comment:
Huh. I swear I had a check for the user role being admin above the get_dataset call, but something happened to it. Maybe that was the bool version from the other branch I ditched. Anyway, thanks for catching this. If we want to avoid an unnecessary DB query we can do that, though. Like
if user.role == "ADMIN":
    return user
Not a bid deal.

Reply:
Yes, I noticed the same problem in my change.  As Daniel said, I put

if user.role == "ADMIN":
   return

above line 34

Reply:
oh, I guess it didn't merge to the `data_manager` branch for some reason, I can add it before the DB